### PR TITLE
Refactor user related flow

### PR DESCRIPTION
**Pull request by yuliaveres**

### CODE REVIEW

**Comment by yuliaveres** [comment]
Verification status loss fix included.

File: Token/APIClients/idAPIClient.swift [hunk 1]
```diff
@@ -105,8 +105,8 @@ public class IDAPIClient: NSObject, CacheExpiryDefault {
 
     public func registerUserIfNeeded(_ success: @escaping (() -> Void)) {
         self.retrieveUser(username: Cereal.shared.address) { user in
-            guard user == nil else {
-                TokenUser.current = user
+            if let user = user as TokenUser? {
+                TokenUser.createOrUpdateCurrentUser(with: user.asDict)
```
Comment (Elland) [inline Token/APIClients/idAPIClient.swift]: Nice!

File: Token/APIClients/idAPIClient.swift [hunk 2]
```diff
@@ -130,7 +130,10 @@ public class IDAPIClient: NSObject, CacheExpiryDefault {
                         guard response.statusCode == 200 else { return }
                         guard let json = json?.dictionary else { return }
 
-                        TokenUser.current = TokenUser(json: json)
+                        TokenUser.createOrUpdateCurrentUser(with: json)
+                        
+                        print("Current user verified: \(String(describing: TokenUser.current?.verified))")
```
Comment (Elland) [inline Token/APIClients/idAPIClient.swift]: `Current user is verified:`?

File: Token/APIClients/idAPIClient.swift [hunk 3]
```diff
@@ -192,7 +195,7 @@ public class IDAPIClient: NSObject, CacheExpiryDefault {
                     guard response.statusCode == 200 else { fatalError() }
                     guard let json = json?.dictionary else { fatalError() }
 
-                    TokenUser.current?.update(json: json, updateAvatar: false, shouldSave: true)
+                    TokenUser.current?.update(json: json)
```
Comment (Elland) [inline Token/APIClients/idAPIClient.swift]: 💪🏽

File: Token/Models/TokenUser.swift [hunk 4]
```diff
@@ -266,6 +240,49 @@ public class TokenUser: NSObject, NSCoding {
...
+        
+        self.current?.update(json: json)
+    }
+    
+    public static func retrieveCurrentUser() {
+        self.current = self.retrieveCurrentUserFromStore()
+    }
+}
+
+private extension TokenUser {
```
Comment (Elland) [inline Token/Models/TokenUser.swift]: Why is this an extension? We only use them to conform to protocols so far.

File: Token/APIClients/idAPIClient.swift (hunk 2 above)
Comment (yuliaveres) [inline Token/APIClients/idAPIClient.swift]: no problem :D

File: Token/Models/TokenUser.swift (hunk 4 above)
Comment (yuliaveres) [inline Token/Models/TokenUser.swift]: i though to try keep private api separately 🤔

Comment (Elland) [inline Token/Models/TokenUser.swift]: I think they're already separated by being private. Just keep them all together inside the object. 💪🏽

Comment (yuliaveres) [inline Token/Models/TokenUser.swift]: Okay ..

**Comment by Elland** [comment]
Running solid here. 🚀

**Pull request merged**